CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+
+All new contributions must also be released under this license.
+
+### Code of Conduct
+
+By contributing you agree to the [code of conduct](CODE_OF_CONDUCT.md). We
+follow the open source best practice of using the [Contributor
+Covenant](https://www.contributor-covenant.org/) for our Code of Conduct.
+
+### Sign your work
+
+To improve tracking of who did what, we follow the Linux Kernel's
+["sign your work" system](https://github.com/wking/signed-off-by).
+This is also called a
+["DCO" or "Developer's Certificate of Origin"](https://developercertificate.org/).
+
+**All** commits are required to include this sign off and we use the
+[Probot DCO App](https://github.com/probot/dco) to check pull requests for
+this.
+
+The sign-off is a simple line at the end of the explanation for the
+patch, which certifies that you wrote it or otherwise have the right to
+pass it on as a open-source patch.  The rules are pretty simple: if you
+can certify the below:
+
+        Developer's Certificate of Origin 1.1
+
+        By making a contribution to this project, I certify that:
+
+        (a) The contribution was created in whole or in part by me and I
+            have the right to submit it under the open source license
+            indicated in the file; or
+
+        (b) The contribution is based upon previous work that, to the best
+            of my knowledge, is covered under an appropriate open source
+            license and I have the right under that license to submit that
+            work with modifications, whether created in whole or in part
+            by me, under the same open source license (unless I am
+            permitted to submit under a different license), as indicated
+            in the file; or
+
+        (c) The contribution was provided directly to me by some other
+            person who certified (a), (b) or (c) and I have not modified
+            it.
+
+	(d) I understand and agree that this project and the contribution
+	    are public and that a record of the contribution (including all
+	    personal information I submit with it, including my sign-off) is
+	    maintained indefinitely and may be redistributed consistent with
+	    this project or the open source license(s) involved.
+
+then you just add a line saying
+
+	Signed-off-by: Random J Developer <random@developer.example.org>
+
+using your real name (sorry, no pseudonyms or anonymous contributions.)
+
+You can add the signoff as part of your commit statement. For example:
+
+    git commit --signoff -a -m "Fixed some errors."
+
+*Hint:* If you've forgotten to add a signoff to one or more commits, you can use the
+following command to add signoffs to all commits between you and the upstream
+master:
+
+    git rebase --signoff upstream/master
+
+
+----
+
+This file is generated from [README.md](README.md), please edit that file then
+run the `./.github/update-contributing.py`.
```

Comment:
What is meant by these lines? Also, which repo are you pulling from mainly for this file?

Reply:
This example is used in a few places I believe, but I took it mainly from the project Xray repo. In that repo they just put the Contributing info at the end of the readme, and then run the python script to publish that specific section under CONTRIBUTING.md. Honestly, it seems like more of a hassle to me, but if the Contributing info needs to be included in the readme, this is probably the easiest way of doing it. 

I'll look around at some of the other repos to see if they all have contributing info in the readme as well. If not, we can just delete these lines.

Reply:
sounds good